websites/T/TOD/presence.ts

```diff
@@ -0,0 +1,296 @@
+class TOD extends Presence {
+	constructor(presenceOptions: PresenceOptions) {
+		super(presenceOptions);
+	}
+
+	getVideo() {
+		return document.querySelector<HTMLMediaElement>("video");
+	}
+
+	getVideoType() {
+		if (
+			this.getVideoTitle().match(/S\d+\s*\|\s*E\d+/i) ||
+			this.getVideoTitle().match(/EP\s*\d+/i)
+		)
+			return "show";
+		else return "movie";
+	}
+
+	getTitle(eyebrow = false) {
+		if (this.isWatching() && eyebrow) return this.getVideoTitle();
+
+		const title = document.querySelector(
+			"div.bein-dh1-hero__head-container > h1.bein-dh1-hero__title.heading-shadow"
+		)?.textContent;
+
+		return (
+			title ??
+			this.getVideoTitle().replace(/\d+\s*-\s*\d+/ as RegExp, "vs") ??
+			"Unknown"
+		);
+	}
+
+	getVideoTitle() {
+		return document.querySelector("div.diva-standard-title")?.textContent;
+	}
+
+	isWatching() {
+		return !!this.getVideoTitle();
+	}
+
+	isLive() {
+		return !!document.querySelector("span.diva-live-now");
+	}
+
+	isTrailer() {
+		return this.getVideoTitle().includes("Trailer");
+	}
+}
+
+const enum Assets {
+	Logo = "https://i.imgur.com/2euETKm.png",
+}
+
+const presence = new TOD({
+		clientId: "1229087342787039334",
+	}),
+	data: {
+		startedSince: number;
+		settings?: {
+			id: string;
+			delete?: boolean;
+			data: string[];
+		}[];
+		presence: {
+			[key: string]: {
+				setPresenceData?: () => void;
+			};
+		};
+	} = {
+		presence: {},
+		startedSince: Math.floor(Date.now() / 1000),
+	};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		type: ActivityType.Watching,
+		smallImageKey: Assets.Search,
+		smallImageText: "Browsing",
+		largeImageKey: Assets.Logo,
+		details: "Browsing...",
+		startTimestamp: data.startedSince,
+	};
+
+	data.presence = {
+		"/movie/([a-zA-Z0-9-]+)": {
+			setPresenceData() {
+				if (presence.isWatching()) {
+					const video = presence.getVideo();
+
+					[, presenceData.endTimestamp] =
+						presence.getTimestampsfromMedia(video);
+
+					if (presence.isTrailer()) {
+						presenceData.details = "Watching a trailer";
+						presenceData.state = presence.getVideoTitle().replace(/- (.+)/, "");
+
+						presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+						presenceData.smallImageKey = video.paused
+							? Assets.Pause
+							: Assets.Play;
+
+						presenceData.buttons = [
+							{
+								label: "Watch Trailer",
+								url: document.location.href,
+							},
+						];
+					} else {
+						presenceData.details = `Watching ${presence.getVideoType()}`;
+						presenceData.state = presence.getVideoTitle();
+
+						presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+						presenceData.smallImageKey = video.paused
+							? Assets.Pause
+							: Assets.Play;
+
+						presenceData.buttons = [
+							{
+								label: `Watch ${
+									presence.getVideoType() === "show"
+										? "Show"
+										: presence.getVideoType() === "movie"
+										? "Movie"
+										: "Unknown"
```

Comment:
`Unknown` would never occur as the video type is either "show" or "movie"
```suggestion
									presence.getVideoType() === "show"
										? "Show"
										: "Movie"
```